### PR TITLE
Signal handling reworked.

### PR DESCRIPTION
**Pull request by shindo**
Merge #30 first.

All signal handling mechanics moved to thevoid::signal_handler namespace.

Added functions to add/remove server to/from global signal handling:
    thevoid::signal_handler::add_server(base_server\*)
    thevoid::signal_handler::remove_server(base_server\*)

Added functions to register stop/reload signals:
    thevoid::signal_handler::register_stop(int signal)
    thevoid::signal_handler::register_reload(int signal)

### CODE REVIEW

**Comment by shindo** [comment]
signal_handler's internal functions and data will be moved to unnamed namespace on #32.

File: thevoid/server.cpp [hunk 1]
```diff
@@ -50,9 +50,62 @@
...
+namespace signal_handler {
 
-static std::weak_ptr<signal_handler> global_signal_set;
+std::mutex lock;
+std::set<base_server*> all_servers;
+
+static
+void stop(int signal_value)
+{
+	std::lock_guard<std::mutex> locker(lock);
```
Comment (bioothod) [inline thevoid/server.cpp]: It is strictly forbidden to use any kind of locks in signal handlers.
Pull requested is closed.

Comment (shindo) [inline thevoid/server.cpp]: Really? Have you seen the current implementation?

Comment (bioothod) [inline thevoid/server.cpp]: Having a bug is not an excuse to create another one.
You should fix bugs, not copy them around.

Comment (euroelessar) [inline thevoid/server.cpp]: Danil, try to implement handling of signals using signalfd, you may listen it at monitoring thread. As it's a common file descriptor you'll be allows to do whatever you want including using of mutexes and writing to logger.

**Pull request closed without merging**